test/unit/detail/format_help_test.cpp

```diff
@@ -77,19 +77,32 @@ struct test_accessor
 
 TEST(help_page_printing, short_help)
 {
-    // Empty call with no options given. For sharg::detail::format_short_help
-    sharg::parser parser0{"empty_options", 1, argv_without_any_options};
-    sharg::detail::test_accessor::set_terminal_width(parser0, 80);
-    parser0.info.synopsis.push_back("./some_binary_name synopsis");
-    testing::internal::CaptureStdout();
-    EXPECT_EXIT(parser0.parse(), ::testing::ExitedWithCode(EXIT_SUCCESS), "");
-    std_cout = testing::internal::GetCapturedStdout();
-
     expected = "empty_options\n"
                "=============\n"
                "    ./some_binary_name synopsis\n"
                "    Try -h or --help for more information.\n";
-    EXPECT_EQ(std_cout, expected);
+
+    // Empty call with no options given. For sharg::detail::format_short_help
+    // even if required options exist.
+
+    auto check_expected_short_help = [&](auto argv)
+    {
+        sharg::parser parser{"empty_options", 1, argv};
```

Comment:
I think this should be something else than a `1` but the actual length of the array `argv`

Reply:
> Some snippet: `int const argc = sizeof(argv) / sizeof(*argv);`

That was also my first impulse. But that doesn't work inside the lambda, because argv is an unsized array (no clue what the correct term is), meaning the type of `argv` is not of type `char const* argv[3]`, but of type `char const* argv[]` which inside a function signature is reduced to something similar to `char const** argv`.

https://godbolt.org/z/5TfPMenT4

Reply:
Has to be `(auto && v)`. I guess the type allowed to decay on copy

Reply:
> Has to be `(auto && v)`. I guess the type allowed to decay on copy

Uh, very cool trick 😎, wasn't aware of this!

Reply:
ouch.. thanks for spotting this.